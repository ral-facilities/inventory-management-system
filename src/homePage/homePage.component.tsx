import React from 'react';
import Typography from '@mui/material/Typography';
import { Grid, Box, Paper, Button, alpha, styled } from '@mui/material';
import { Link } from 'react-router-dom';
import DGLogo from '/images/scigateway-white-text-blue-mark-logo.svg';
import BackgroundImage from '/images/background.jpg';
import GreenSwirl1Image from '/images/green-swirl1.png';
import GreenSwirl2Image from '/images/green-swirl2.png';
import Decal1Image from '/images/decal1.svg';
import Decal2Image from '/images/decal2.svg';
import Decal2DarkImage from '/images/decal2-dark.svg';
import Decal2DarkHCImage from '/images/decal2-darkhc.svg';
import FacilityImage from '/images/facility.jpg';
import { InventoryManagementSystemSettingsContext } from '../configProvider.component';

export interface BaseHomePageProps {
  logo: string;
  backgroundImage: string;
  greenSwirl1Image: string;
  greenSwirl2Image: string;
  decal1Image: string;
  decal2Image: string;
  decal2DarkImage: string;
  decal2DarkHCImage: string;
  facilityImage: string;
}

const backgroundTitleStyles = {
  color: '#FFFFFF',
  margin: 'auto',
  fontSize: '48px',
  fontWeight: 'lighter',
  textAlign: 'center',
};

const paperStyles = {
  borderRadius: '4px',
  marginBottom: 2,
  height: '100%',
};

const paperContentStyles = {
  padding: 2,
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-start',
  height: '100%',
  boxSizing: 'border-box',
  textAlign: 'left',
};

const PaperHeading = styled(Typography)(({ theme }) => ({
  fontWeight: 'bold',
  fontSize: '24px',
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  color: (theme as any).colours?.homePage?.heading,
  marginBottom: theme.spacing(2),
}));

const PaperDescription = styled(Typography)(({ theme }) => ({
  textAlign: 'left',
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  color: (theme as any).colours?.contrastGrey,
  marginBottom: theme.spacing(2),
}));

const BluePaperHeading = styled(Typography)(({ theme }) => ({
  fontWeight: 'bold',
  fontSize: '24px',
  color: '#FFFFFF',
  marginBottom: theme.spacing(2),
}));

const BluePaperDescription = styled(Typography)(({ theme }) => ({
  textAlign: 'left',
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  color: (theme as any).colours?.homePage?.blueDescription,
  marginBottom: theme.spacing(2),
}));

interface BrowseDecalProps {
  decal2Image: string;
  decal2DarkImage: string;
  decal2DarkHCImage: string;
}

const BrowseDecal = styled('div', {
  shouldForwardProp: (prop) =>
    prop !== 'decal2Image' &&
    prop !== 'decal2DarkImage' &&
    prop !== 'decal2DarkHCImage',
})<BrowseDecalProps>(
  ({ theme, decal2Image, decal2DarkImage, decal2DarkHCImage }) => ({
    backgroundImage:
      theme.palette.mode === 'light'
        ? `url(${decal2Image})`
        : // eslint-disable-next-line @typescript-eslint/no-explicit-any
          (theme as any).colours?.type === 'default'
          ? `url(${decal2DarkImage})`
          : `url(${decal2DarkHCImage})`,
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'top left',
    backgroundSize: 'auto 100%',
    height: '100%',
  })
);

const LightBlueButton = styled(Button)(({ theme }) => ({
  color: '#FFFFFF',
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  backgroundColor: (theme as any).colours?.homePage?.blueButton,
  '&:hover': {
    //Check if null to avoid error when loading
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    backgroundColor: (theme as any).colours?.homePage?.blueButton
      ? alpha(
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          (theme as any).colours?.homePage?.blueButton,
          0.8
        )
      : '#FFFFFF',
  },
}));

const BaseHomePage = (props: BaseHomePageProps): React.ReactElement => {
  return (
    <div id="dg-homepage">
      <div
        style={{
          backgroundImage: `url(${props.backgroundImage})`,
          backgroundPosition: 'center 40%',
          width: '100%',
          height: 250,
        }}
      >
        <div
          style={{
            backgroundImage: `url(${props.greenSwirl1Image}), url(${props.decal1Image})`,
            backgroundRepeat: 'no-repeat',
            backgroundPosition: 'top left, top right',
            width: '100%',
            height: 250,
          }}
        >
          <Box
            sx={{
              position: 'relative',
              left: '50%',
              top: '45px',
              transform: 'translate(-50%)',
            }}
          >
            <Typography variant="h2" sx={backgroundTitleStyles}>
              <strong>Inventory Management</strong>
            </Typography>
            <Typography variant="h2" sx={backgroundTitleStyles}>
              for <strong>EPAC</strong>
            </Typography>
          </Box>
        </div>
      </div>
      <Box
        sx={{
          transform: 'translate(0px, -20px)',
          marginLeft: '8%',
          marginRight: '8%',
        }}
      >
        <Paper sx={paperStyles} elevation={1}>
          <Grid container style={{ height: '100%' }}>
            <Grid item xs={6}>
              <Box sx={paperContentStyles}>
                <Typography
                  variant="h3"
                  sx={(theme) => ({
                    fontWeight: 'bold',
                    fontSize: '32px',
                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
                    color: (theme as any).colours?.homePage?.heading,
                    marginBottom: theme.spacing(2),
                  })}
                >
                  {'Catalogue'}
                </Typography>
                <PaperDescription variant="body1">
                  {
                    'Browse catalogue categories, sub categories and catalogue items to locate specific items or add new ones.'
                  }
                </PaperDescription>
                <Box marginTop="16px">
                  <Button
                    sx={(theme) => ({
                      // eslint-disable-next-line @typescript-eslint/no-explicit-any
                      backgroundColor: (theme as any).colours?.darkBlue,
                    })}
                    color="primary"
                    variant="contained"
                    component={Link}
                    to={'/catalogue'}
                    data-testid="catalogue-button"
                  >
                    {'Explore'}
                  </Button>
                </Box>
              </Box>
            </Grid>
            <Grid item xs={6}>
              <div
                style={{
                  backgroundImage: `url(${props.facilityImage})`,
                  backgroundRepeat: 'no-repeat',
                  backgroundPosition: 'bottom right',
                  backgroundSize: 'cover',
                  width: '100%',
                  height: '100%',
                  borderRadius: '4px',
                }}
              >
                <BrowseDecal
                  decal2Image={props.decal2Image}
                  decal2DarkImage={props.decal2DarkImage}
                  decal2DarkHCImage={props.decal2DarkHCImage}
                />
              </div>
            </Grid>
          </Grid>
        </Paper>
        <Grid container spacing={2}>
          <Grid item sm={12} md={4}>
            <Paper sx={paperStyles} elevation={1}>
              <Box sx={paperContentStyles}>
                <PaperHeading variant="h4">{'Systems'}</PaperHeading>
                <PaperDescription variant="body1">
                  {
                    'Navigate through systems and their sub systems to view their component items.'
                  }
                </PaperDescription>
                <Box marginTop="auto">
                  <Button
                    sx={(theme) => ({
                      // eslint-disable-next-line @typescript-eslint/no-explicit-any
                      backgroundColor: (theme as any).colours?.darkBlue,
                    })}
                    color="primary"
                    variant="contained"
                    component={Link}
                    to={'/systems'}
                    data-testid="systems-button"
                  >
                    {'Explore'}
                  </Button>
                </Box>
              </Box>
            </Paper>
          </Grid>
          <Grid item sm={12} md={4}>
            <Paper sx={paperStyles} elevation={1}>
              <Box sx={paperContentStyles}>
                <PaperHeading variant="h4">{'Manufacturers'}</PaperHeading>
                <PaperDescription variant="body1">
                  {'View the manufacturers of parts used by EPAC.'}
                </PaperDescription>
                <Box marginTop="auto">
                  <Button
                    sx={(theme) => ({
                      // eslint-disable-next-line @typescript-eslint/no-explicit-any
                      backgroundColor: (theme as any).colours?.darkBlue,
                    })}
                    color="primary"
                    variant="contained"
                    component={Link}
                    to={'/manufacturers'}
                    data-testid="manufacturer-button"
                  >
                    {'Explore'}
                  </Button>
                </Box>
              </Box>
            </Paper>
          </Grid>
          <Grid item sm={12} md={4}>
            <Paper
              sx={{ ...paperStyles, backgroundColor: '#003088' }}
              elevation={1}
            >
              <div
                style={{
                  backgroundImage: `url(${props.greenSwirl2Image})`,
                  backgroundRepeat: 'no-repeat',
                  backgroundPosition: 'top right',
                  backgroundSize: 'auto 100%',
                  height: '100%',
                }}
              >
                <Box sx={paperContentStyles}>
                  <BluePaperHeading variant="h4">{'EPAC'}</BluePaperHeading>
                  <BluePaperDescription color="#ffffff" variant="body1">
                    {
                      'World-leading centre for scientific and industrial research.'
                    }
                  </BluePaperDescription>
                  <Box marginTop="auto">
                    <LightBlueButton
                      color="primary"
                      variant="contained"
                      href={
                        'https://www.clf.stfc.ac.uk/Pages/EPAC-Applications.aspx'
                      }
                      data-testid="facility-button"
                    >
                      {'Read more'}
                    </LightBlueButton>
                  </Box>
                </Box>
              </div>
            </Paper>
          </Grid>
        </Grid>
      </Box>
    </div>
  );
};

export const HomePage = React.memo((): React.ReactElement => {
  const settings = React.useContext(InventoryManagementSystemSettingsContext);
  const pluginHost = settings.pluginHost;

  const HomePage = BaseHomePage;
  return (
    <HomePage
      logo={pluginHost + DGLogo}
      backgroundImage={pluginHost + BackgroundImage}
      greenSwirl1Image={pluginHost + GreenSwirl1Image}
      greenSwirl2Image={pluginHost + GreenSwirl2Image}
      decal1Image={pluginHost + Decal1Image}
      decal2Image={pluginHost + Decal2Image}
      decal2DarkImage={pluginHost + Decal2DarkImage}
      decal2DarkHCImage={pluginHost + Decal2DarkHCImage}
      facilityImage={pluginHost + FacilityImage}
    />
  );
});
HomePage.displayName = 'HomePage';
