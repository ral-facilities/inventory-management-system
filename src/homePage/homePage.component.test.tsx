import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { HomePage } from '../homePage/homePage.component';

describe('Home page component', () => {
  it('homepage renders correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
