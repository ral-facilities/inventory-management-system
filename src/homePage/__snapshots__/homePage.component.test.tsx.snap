// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page Error Component > renders catalogue error page correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiBox-root css-d8r78j"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1wffj3i-MuiSvgIcon-root"
        data-testid="BugReportIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20zm-6 8h-4v-2h4zm0-4h-4v-2h4z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1fjbu-MuiTypography-root"
      >
        404
      </p>
    </div>
    <div
      class="MuiBox-root css-sg7ujc"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-18ft2kn-MuiTypography-root"
      >
        Page not found
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 css-n3u125-MuiTypography-root"
      >
        We're sorry, the page you requested was not found on the server. If you entered the URL manually please check your spelling and try again. Otherwise, return to the 
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-17fxp5y-MuiTypography-root-MuiLink-root"
          href="/"
        >
           home page
        </a>
         .
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home page component > homepage renders correctly 1`] = `
<DocumentFragment>
  <div
    id="dg-homepage"
  >
    <div
      style="background-image: url("/images/background.jpg"); width: 100%; height: 250px;"
    >
      <div
        style="background-repeat: no-repeat; width: 100%; height: 250px;"
      >
        <div
          class="MuiBox-root css-h95k7f"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1c7qyh0-MuiTypography-root"
          >
            <strong>
              Inventory Management
            </strong>
          </h2>
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1c7qyh0-MuiTypography-root"
          >
            for 
            <strong>
              EPAC
            </strong>
          </h2>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-12vlygr"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-177u6i-MuiPaper-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          style="height: 100%;"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-bvoom2"
            >
              <h3
                class="MuiTypography-root MuiTypography-h3 css-jadg01-MuiTypography-root"
              >
                Catalogue
              </h3>
              <p
                class="MuiTypography-root MuiTypography-body1 css-16w7kz2-MuiTypography-root"
              >
                Browse catalogue categories, sub categories and catalogue items to locate specific items or add new ones.
              </p>
              <div
                class="MuiBox-root css-1yuhvjn"
              >
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
                  data-testid="catalogue-button"
                  href="/catalogue"
                  tabindex="0"
                >
                  Explore
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div
              style="background-image: url("/images/facility.jpg"); background-repeat: no-repeat; background-position: bottom right; background-size: cover; width: 100%; height: 100%; border-radius: 4px;"
            >
              <div
                class="css-1ox020v"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-yaog9p-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-177u6i-MuiPaper-root"
          >
            <div
              class="MuiBox-root css-bvoom2"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 css-rspydk-MuiTypography-root"
              >
                Systems
              </h4>
              <p
                class="MuiTypography-root MuiTypography-body1 css-16w7kz2-MuiTypography-root"
              >
                Navigate through systems and their sub systems to view their component items.
              </p>
              <div
                class="MuiBox-root css-8xl60i"
              >
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
                  data-testid="systems-button"
                  href="/systems"
                  tabindex="0"
                >
                  Explore
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-yaog9p-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-177u6i-MuiPaper-root"
          >
            <div
              class="MuiBox-root css-bvoom2"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 css-rspydk-MuiTypography-root"
              >
                Manufacturers
              </h4>
              <p
                class="MuiTypography-root MuiTypography-body1 css-16w7kz2-MuiTypography-root"
              >
                View the manufacturers of parts used by EPAC.
              </p>
              <div
                class="MuiBox-root css-8xl60i"
              >
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
                  data-testid="manufacturer-button"
                  href="/manufacturers"
                  tabindex="0"
                >
                  Explore
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-4 css-yaog9p-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-19bggzh-MuiPaper-root"
          >
            <div
              style="background-image: url("/images/green-swirl2.png"); background-repeat: no-repeat; background-position: top right; background-size: auto 100%; height: 100%;"
            >
              <div
                class="MuiBox-root css-bvoom2"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-94tke6-MuiTypography-root"
                >
                  EPAC
                </h4>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1kje51w-MuiTypography-root"
                >
                  World-leading centre for scientific and industrial research.
                </p>
                <div
                  class="MuiBox-root css-8xl60i"
                >
                  <a
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1m6sz6p-MuiButtonBase-root-MuiButton-root"
                    data-testid="facility-button"
                    href="https://www.clf.stfc.ac.uk/Pages/EPAC-Applications.aspx"
                    tabindex="0"
                  >
                    Read more
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
